firmware/STN1110.c

```diff
@@ -29,56 +29,102 @@
 #include <stdlib.h>
 #include "settings.h"
 #include "system.h"
+#include "modp_numtoa.h"
 
 #define LOG_PFX "SYS_STN1110: "
 
-#define STR_HELPER(x) #x
-#define STR(x) STR_HELPER(x)
-#define STN1110_RUNTIME_BAUD_RATE 230400
+#define RESET_DELAY 10
+#define AT_COMMAND_DELAY 100
+#define LONG_DELAY 1000
 
 /* Receive Buffer for the STN1110 */
 char stn_rx_buf[1024];
 
-
+/* Send an AT command */
 static void _send_at(char *at_cmd)
 {
     sdWrite(&SD2, (uint8_t*)at_cmd, strlen(at_cmd));
-    chThdSleepMilliseconds(100);
+    sdWrite(&SD2, (uint8_t*)"\r", 1);
+    chThdSleepMilliseconds(AT_COMMAND_DELAY);
+}
+
+/* Send an AT command with a single numberic parameter value */
+static void _send_at_param(char *at_cmd, int param)
+{
+    sdWrite(&SD2, (uint8_t*)at_cmd, strlen(at_cmd));
+    char param_str[20];
```

Comment:
Max len an int will ever be is 14 characters (12 digits + negative or positive sign + NULL).  So 16 could save you some stack